pyo3-ffi/src/lib.rs

```diff
@@ -265,6 +265,19 @@ macro_rules! opaque_struct {
     };
 }
 
+macro_rules! addr_of_mut_shim {
```

Comment:
Short answer: It's a shim, so let's call it a shim 🤔

Long answer: It can work with the same name, but I'd rather not pretend it's the actual `addr_of_mut`. If it was named that the casual reader wouldn't be able to tell that it's not `split_at_mut`.

Reply:
> If it was named that the casual reader wouldn't be able to tell that it's not addr_of_mut.

That was my actual intent: Not having to think about the shim part while reading the code making use of it. (Also under the assumption that the shim will be removed once the MSRV passes 1.51.)

But fair enough calling it a shim if that is considered sufficiently important.

Reply:
> Also under the assumption that the shim will be removed once the MSRV passes 1.51.

Absolutely, there's quite of bit of addr_of/min_const_generics stuff that can be deleted with a 1.51 msrv 🎉

Reply:
I also wondered about just calling the macro `addr_of_mut`; we can always be explicit and use `crate::addr_of_mut` everywhere to make it obvious that it's different from the stdlib if we wanted. Calling it `addr_of_mut_shim` is also fine.

I'm already looking forward to the next MSRV / Python bump; if we wait until mid 2023 when 3.7 reaches end of life we might get quite a few upgrades :)